Microsoft.Toolkit.Uwp/Helpers/DeepLinkParser.cs

```diff
@@ -0,0 +1,142 @@
+    ///         }
+    /// </code>
+    /// </example>
+    public class DeepLinkParser : Dictionary<string, string>
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="DeepLinkParser"/> for the given <see cref="IActivatedEventArgs"/>
+        /// </summary>
+        /// <param name="args">The <see cref="IActivatedEventArgs"/> instance containing the launch Uri data.</param>
+        /// <returns>An instance of <see cref="DeepLinkParser"/></returns>
+        /// <remarks><paramref name="args"/> will be cast to <see cref="ILaunchActivatedEventArgs"/> </remarks>
+        public static DeepLinkParser Create(IActivatedEventArgs args) => new DeepLinkParser(args);
+
+        /// <summary>
+        /// Validates the source URI.
+        /// </summary>
+        /// <param name="uri">The URI.</param>
+        /// <returns>
+        ///   <paramref name="uri" /> as a <c>System.Uri</c> instance
+        /// </returns>
+        /// <exception cref="System.ArgumentNullException">thrown if <paramref name="uri"/> is null</exception>
+        /// <exception cref="System.ArgumentException">Not a valid URI format</exception>
+        protected static Uri ValidateSourceUri(string uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+
+            Uri validatedUri;
+            if (!Uri.TryCreate(uri, UriKind.RelativeOrAbsolute, out validatedUri)
+                || !validatedUri.IsWellFormedOriginalString())
+            {
+                throw new ArgumentException("Not a valid URI format", nameof(uri));
+            }
+
+            return validatedUri;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeepLinkParser"/> class.
+        /// </summary>
+        protected DeepLinkParser()
+        {
+        }
+
+        private DeepLinkParser(IActivatedEventArgs args)
+        {
+            var inputArgs = args as ILaunchActivatedEventArgs;
+
+            if (inputArgs == null)
+            {
+                throw new ArgumentException("'args' is not a LaunchActivatedEventArgs instance", nameof(args));
+            }
+
+            ParseUriString(inputArgs.Arguments);
+        }
+
+        /// <summary>
+        /// Parses the URI string in to components.
+        /// </summary>
+        /// <param name="uri">The URI.</param>
+        protected virtual void ParseUriString(string uri)
+        {
+            Uri validatedUri = ValidateSourceUri(uri);
+
+            string queryString = SetRootAndGetQueryString(validatedUri);
+            if (!string.IsNullOrWhiteSpace(queryString))
+            {
+                foreach (var queryStringParam in queryString.Split('&')
+                    .Select(param =>
+                    {
+                        var kvp = param.Split('=');
+                        return new KeyValuePair<string, string>(kvp[0], kvp[1]);
```

Comment:
You've essentially done this conversion to KeyValuePair list twice now, could we move this to a helper method instead?